src/client/index.ts

```diff
@@ -297,6 +312,14 @@ export class Client {
 
 	async negotiate(issuers?: (OnChainTokenConfig | OffChainTokenConfig)[], openPopup = false) {
 
+		let currentIssuer = this.getOutletConfigForCurrentOrigin();
```

Comment:
@oleggrib Am I correct in thinking that we no longer need to initialize Outlet on pages where Client is used as well? For instance, I can just initialize client and call negotiate and magic link will be read?

Reply:
right, this is why I did it, because developer can forget to add Outlet code and MagicLInk save can be broken, same as redirect flow. @micwallace

Reply:
Hi guys. 

I think it's a good idea to make the developers life easier. However by allowing this, what's the outcome if the developer does initialise the outlet on a { client, outlet } configuration?

Is it not best to keep this items seperate for now and optimise later?

Thanks guys, open to the direction we take my side.

Reply:
@oleggrib @nicktaras I like the idea of having outlet initialised by negotiator for same-origin outlets. 

However there are a few considerations:
1. Do we really need to initialise the whole outlet instance just to read a magic link. 
2. I think magic link reading should be separated from outlet to reduce complexity, maybe a static method that can be called within negotiator, outlet or even directly.
3. From looking at the diff, I can't see the LocalOutlet class being used to read the tokens direct from local storage. Maybe I am just tired LOL

Reply:
Hi guys, I'm going to close these conversations.

Thanks @micwallace - you're right, we should look to break down the code / optimise further. Where we could have a module named - { magicLink } to include in examples where this is required to lighten the code inside outlet.

For now, let us first focus on the delivery of BC V2, and park up refactoring until later / unless needed during this development phase.